Yank/repex.py

```diff
@@ -680,30 +806,29 @@ def write_timestamp(self, iteration):
                 storage.createVariable('timestamp', str, ('iteration',), zlib=False, chunksizes=(1,))
         timestamp = time.ctime()
         self._storage_analysis.variables['timestamp'][iteration] = timestamp
-        checkpoint_interval = self._calculate_checkpoint_iteration(iteration)
-        if checkpoint_interval:
-            self._storage_checkpoint.variables['timestamp'][iteration] = timestamp
+        checkpoint_iteration = self._calculate_checkpoint_iteration(iteration)
+        if checkpoint_iteration is not None:
+            self._storage_checkpoint.variables['timestamp'][checkpoint_iteration] = timestamp
 
-    def read_dict(self, name, storage='analysis'):
+    def read_dict(self, name):
         """Restore a dictionary from the storage file.
 
         Parameters
         ----------
         name : str
             The identifier of the dictionary used to stored the data.
-        storage : 'analysis' or 'checkpoint', optional, Default: 'analysis'
-            Stores a dictionary on the given storage file.
-            'analysis': Frequently written/accessed data, usually smaller in size
-                e.g. energies, state indices, etc.
-            'checkpoint': Infrequent written/accessed data, usually larger and needed to restore from a checkpoint
-                e.g. Explicit solvent coordinates, serialized state string, simulation options.
 
         Returns
         -------
         data : dict
             The restored data as a dict.
 
         """
+        # Leaving the skeleton to extend this in for now
+        if name in ['metadata']:  # store data needed for analysis in the primary file
+            storage = 'analysis'
+        else:
+            storage = 'checkpoint'
```

Comment:
Sorry if I wasn't clear, what I mean is that many individual `MCMCMove` objects store their own `n_accepted` and `n_proposed` variables (e.g. `GHMCMove`, `MCRotationMove`) that I'm sure we'll want to read from the analysis soon in the future.